src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1276,170 +1249,411 @@ end subroutine eigencalving
+               calvingThickness(iCell) = thickness(iCell)
+               calvingSubtotal3 = calvingSubtotal3 + calvingThickness(iCell)
+            endif
+         endif
+         if (isnan(areaCell(iCell))) then
+            call mpas_log_write("NaN detected in calvingThickness at cell $i", MPAS_LOG_ERR, intArgs=(/iCell/))
+            err_tmp = 1
+            err = ior(err, err_tmp)
+         endif
+      enddo
 
-         call mpas_log_write("distribute_calving_flux failed to distribute all required ice - " // &
-                    " ice was left after depleting all neighbors." // &
-                    "  Search needs to be expanded to neighbors' neighbors.")
-         call mpas_log_write("   On this processor: $i cells contain uncalved ice, " // &
-                 "for a total uncalved volume of $r m^3 ($r%).", &
-                 MPAS_LOG_WARN, intArgs=(/uncalvedCount/), &
-                 realArgs=(/uncalvedTotal, 100.0_RKIND * uncalvedTotal/(requiredCalvingVolumeRate(iCell) * deltat)/))
+
+      ! End of routine accounting/reporting
+      localInfo(1) = calvingSubtotal1
+      localInfo(2) = calvingSubtotal2
+      localInfo(3) = calvingSubtotal3
+      localInfo(4) = sum(calvingThickness(1:nCellsSolve) * areaCell(1:nCellsSolve))
+      localInfo(5) = sum(uncalvedVolumeNonDynCell(1:nCellsSolve))
+      localInfo(6) = sum(uncalvedVolumeDynCell(1:nCellsSolve))
+      ! NOTE: THIS WILL NOT WORK ON MULTIPLE BLOCKS PER PROCESSOR
+      call mpas_dmpar_sum_real_array(domain % dminfo, 6, localInfo, globalInfo)
+      call mpas_log_write("== Calving complete. Total calved                    = $r", realArgs = (/globalInfo(4)/))
+      call mpas_log_write("==                   Calved from non-dynamic cells   = $r", realArgs = (/globalInfo(1)/))
+      call mpas_log_write("==                   Calved from dynamic cells       = $r", realArgs = (/globalInfo(2)/))
+      call mpas_log_write("==                   Stranded floating cells deleted = $r", realArgs = (/globalInfo(3)/))
+      call mpas_log_write("== Uncalved volumes: Non-dynamic cells=$r; Dynamic cells=$r", realArgs=(/globalInfo(5),globalInfo(6)/))
+      if (((globalInfo(5) + globalInfo(6)) / (globalInfo(4) + 1.0e-30_RKIND) > 0.001_RKIND) .and. &
+          (globalInfo(4) > 1000.0_RKIND**2)) then ! Include some small amount of total calving for comparison
+         call mpas_log_write("Calving scheme failed to calve $r m^3. ($r% of total calved)", MPAS_LOG_WARN, &
+                 realArgs=(/globalInfo(5) + globalInfo(6), &
+                 100.0_RKIND * (globalInfo(5) + globalInfo(6)) / (globalInfo(4)+1.0e-30_RKIND)/))
+      endif
+      if (((globalInfo(5) + globalInfo(6)) / (globalInfo(4) + 1.0e-30_RKIND) > 0.1_RKIND) .and. &
+          (globalInfo(4) > 1000.0_RKIND**2)) then ! Include some small amount of total calving for comparison
+         call mpas_log_write("Calving scheme failed to calve an amount greater than 10% of the ice calved.  " // &
```

Comment:
I was wondering what the case is for calved amount is < 10% here. Do we have accumulated uncalved volume over time?

Reply:
This is checking what fraction of volume was uncalved on this timestep.  10% seemed like a reasonable accuracy threshold to tolerate.

Reply:
Would it make sense to add a namelist option to control this threshold?

Reply:
I think the 10% threshold is a good guardrail for now.  As we use the calving routine more, we can adjust it as needed.